app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/add_tools/remote_mcp/+page.svelte

```diff
@@ -0,0 +1,252 @@
+<script lang="ts">
+  import AppPage from "../../../../../app_page.svelte"
+  import FormContainer from "$lib/utils/form_container.svelte"
+  import FormElement from "$lib/utils/form_element.svelte"
+  import FormList from "$lib/utils/form_list.svelte"
+  import { client } from "$lib/api_client"
+  import { page } from "$app/stores"
+  import { goto } from "$app/navigation"
+  import { KilnError, createKilnError } from "$lib/utils/error_handlers"
+
+  // Form fields
+  let name = ""
+  let server_url = ""
+  let description = ""
+
+  // Headers as array of key/value pairs
+  interface HeaderPair {
+    key: string
+    value: string
+    placeholder: string | null
+  }
+
+  let headers: HeaderPair[] = []
+
+  // Form state
+  let error: KilnError | null = null
+  let submitting = false
+
+  // Populate fields from parent page state if provided (only if fields are empty)
+  $: if ($page.state) {
```

Comment:
so a `$` creates a reactive block that will re-run when svelte decides it should (here for page state changing, but should be able to re-run at any time). You wouldn't really want this to re-run reactively, as it could replace user edits with the old static content.

use onMount instead, which runs once after the component mounts.